code/game/machinery/big_manipulator/big_manipulator.dm

```diff
@@ -0,0 +1,909 @@
+/obj/machinery/big_manipulator
+	name = "big manipulator"
+	desc = "Operates different objects. Truly, a groundbreaking innovation..."
+	icon = 'icons/obj/machines/big_manipulator_parts/big_manipulator_core.dmi'
+	icon_state = "core"
+	density = TRUE
+	circuit = /obj/item/circuitboard/machine/big_manipulator
+	greyscale_colors = "#d8ce13"
+	greyscale_config = /datum/greyscale_config/big_manipulator
+	hud_possible = list(BIG_MANIP_HUD)
+
+	/// Is the manipulator turned on?
+	var/on = FALSE
+	/// Is a cycle timer already running?
+	var/cycle_timer_running = FALSE
+
+	/// How quickly the manipulator will process it's actions.
+	var/speed_multiplier = 1
+	var/min_speed_multiplier =  MIN_SPEED_MULTIPLIER_TIER_1
+	var/max_speed_multiplier =  MAX_SPEED_MULTIPLIER_TIER_1
+
+	/// The current task.
+	var/current_task = CURRENT_TASK_NONE
+	var/current_task_start_time = 0
+	var/current_task_duration = 0
+
+	/// The object inside the manipulator.
+	var/datum/weakref/held_object = null
+	/// The chimp worker that uses the manipulator (handles USE cases).
+	var/datum/weakref/monkey_worker = null
+	/// Weakref to the ID that locked this manipulator.
+	var/datum/weakref/id_lock = null
+	/// The manipulator's arm.
+	var/obj/effect/big_manipulator_arm/manipulator_arm = null
+	/// Is the power access wire cut? Disables the power button if `TRUE`.
+	var/power_access_wire_cut = FALSE
+
+	/// How many interaction points of each kind can we have?
+	var/interaction_point_limit = MAX_INTERACTION_POINTS_TIER_1
+	/// List of pickup points.
+	var/list/pickup_points = list()
+	/// List of dropoff points.
+	var/list/dropoff_points = list()
+
+	/// Which tasking scenario we use for pickup points?
+	var/pickup_tasking = TASKING_ROUND_ROBIN
+	/// Which tasking scenario we use for dropoff points?
+	var/dropoff_tasking = TASKING_ROUND_ROBIN
+
+	/// List of all HUD icons resembling interaction points.
+	var/list/hud_points = list()
+
+	/// Pickup strategy for tasking.
+	var/datum/tasking_strategy/pickup_strategy
+	/// Dropoff strategy for tasking.
+	var/datum/tasking_strategy/dropoff_strategy
+
+/obj/machinery/big_manipulator/proc/update_hud()
+  LAZYCLEARLIST(hud_points)
+
+  var/image/main_hud = hud_list[BIG_MANIP_HUD]
+  if(!main_hud)
+    return
+
+  main_hud.loc = get_turf(src)
+  main_hud.appearance = mutable_appearance('icons/effects/interaction_points.dmi', null, ABOVE_NORMAL_TURF_LAYER, src, src.plane)
+
+  main_hud.overlays.Cut()
+  var/list/point_overlays = list()
```

Comment:
why are we operating in this manner? the images list on clients is static right? so like, we still need to make a new image this seems needlessly messy?